### PR TITLE
reposync: catch local file not found urlgrabber error properly (bsc#1208288)

### PR DESCRIPTION
**Pull request by meaksh**
## What does this PR change?

In `urlgrabber` 4.1.0, current version used in Uyuni, there is a small bug [1] on `urlgrab` method that makes the expected "local file not found" exception not to be triggered and therefore producing a different exception in these cases.

Once this small bug is fixed in `urlgrabber`, we need to adapt our current "reposync" code to properly catch the right "local file not found" exception [2] and not produce and error message by default. This is exactly what this PR is doing.

[1] https://github.com/rpm-software-management/urlgrabber/pull/37
[2] https://github.com/rpm-software-management/urlgrabber/blob/master/urlgrabber/grabber.py#L745

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/issues/20616

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: python/test/unit/spacewalk/satellite_tools/test_yum_src.py [hunk 1]
```diff
@@ -54,7 +55,7 @@ def _make_dummy_cs(self):
 
         yum_src.get_proxy = Mock(return_value=(None, None, None))
 
-        cs = yum_src.ContentSource("http://example.com", "test_repo", org='')
+        cs = yum_src.ContentSource("http://example.com/fake_path/", "test_repo", org='')
```
Comment (ycedres) [inline python/test/unit/spacewalk/satellite_tools/test_yum_src.py:58]: Just something I didn't quite understand, why is this needed?

Comment (meaksh) [reply python/test/unit/spacewalk/satellite_tools/test_yum_src.py:58]: I just noticed there was some failures during test execution when using a URL like that, with no real path, as `ContentSource` URL.

There is some code around in reposync and "yum_src" that is apparentely expecting that there is always certain relative path. For example: https://github.com/uyuni-project/uyuni/blame/master/python/spacewalk/satellite_tools/reposync.py#L567

**Review by ycedres — APPROVED** [review]
